src/mappers/country/country_mapper_creator.cpp

```diff
@@ -39,46 +41,137 @@ std::map<std::string, std::string> ImportMappingRules(std::string_view country_m
    return country_mapping_rules;
 }
 
-}  // namespace
-
-
-mappers::CountryMapper mappers::CreateCountryMappings(std::string_view country_mappings_file,
-    const std::map<int, vic3::Country>& countries)
+/// Class so that different mapping strategies can share global variables like country_mappings and
+/// country_mapping_rules
+class CountryMappingCreator
 {
-   std::map<int, std::string> country_mappings;
+   std::map<int, std::string> country_mappings = {};
+   std::map<std::string, std::string> country_mapping_rules;
+   std::vector<const vic3::Country*> deferred_map_countries = {};
 
-   // vic3 tag -> hoi4 tag
-   std::map<std::string, std::string> country_mapping_rules = ImportMappingRules(country_mappings_file);
+   std::set<std::string> used_hoi4_tags = {};
    char tag_prefix = 'Z';
    int tag_suffix = 0;
 
-   std::set<std::string> used_hoi4_tags;
-   for (const vic3::Country& country: countries | std::views::values)
+   // note: these are static because of bound function weirdness
+
+   // Rebel countries should be deferred to give priority to non-rebel countries.
+   static bool DeferCountryWithCivilWar(CountryMappingCreator* this_, const vic3::Country& country)
```

Comment:
1. If possible, make this_ a reference instead of a pointer. Though it doesn't really come up in functions so short, ensure nullability is impossible is good for correctness.
2. Put in-out parameters last in the parameter list.

Reply:
I did it this way because these are "should-be-nonstatic-but-cant-because-of-weird-calling-shenanigans" functions, so I tried to make the `this_` parameter as close as possible to the implicit `this` parameter. It makes the `(**strategy)(this, country)` more explicitly look like a member function call, with all the expectations (parameters of *this may be modified, private methods called, etc.) that that entails.
IDK maybe i've been working too much in assembly and this isn't obvious to normal people :)

Reply:
Just to double-check, you tried them as lambdas?

Reply:
I did not. Do you think this would look better if it were lambdas instead of separate functions?

Reply:
It'd allow you to capture this and save the parameter, at the least. Might also resolve the bound function weirdness, but I'm not very clear on what's happening there or why, so that's uncertain.

Reply:
ugh. lambdas would work except because they're now no longer in class scope, decltype(DeferCountryWithCivilWar) no longer works for other functions so you have to cram everything into one giant function :/

Reply:
(the bound funciton weirdness is that you can't reference a member function via (this->functionname), you have to do CountryMappingCreator::functionname. So the function array goes from 
`{DeferCountryWithCivilWar, AddCountryWithRule, DeferCountryAlways}`
to
`{CountryMappingCreator::DeferCountryWithCivilWar, CountryMappingCreator::AddCountryWithRule, CountryMappingCreator::DeferCountryAlways}`
Which is a lot of extra noise)

Reply:
Gotcha. Annoying they don't work out.

Actually, one last thing to try with them - can you define a more generic std::function interface for them? As in parser.h - https://github.com/ParadoxGameConverters/commonItems/blob/master/Parser.h

So an interface like 
```using CountryStrategyFn= std::function<bool(const vic3::Country&)>;```

Then 
```void ExecuteStrategiesForCountry(const vic3::Country& country, const std::vector<CountryStrategyFn>&& strategies)```
(no pointer)

And finally
```
CountryStrategyFn DeferCountryAlways = [this](const vic3::Country& country) -> bool
   {
      this_->deferred_map_countries_.push_back(&country);
      return true;
   };
```
Or similar syntax.

Reply:
yes mother ;)

Reply:
I decided to not just guess at the syntax but actually try (in case I was wrong). Sent you a PR with it building and passing the tests.